src/CellData/SkeletonCellFieldPair.jl

```diff
@@ -80,12 +76,28 @@ is passed into it. Ideally, if we could parse and extract only the Skeleton
 integration terms from the functional's julia function form, this fix is not
 required, but this is not trivial to do. On the positive side, since the
 evaluations are all lazy and not used, this doesn't put any noticable memory
-or computational overhead.
+or computational overhead. Ofcourse, it is made sure that the such plus side
+pick doesn't happen when the integration over the SkeletonTriangulation
 =#
-function change_domain(a::SkeletonCellFieldPair,target_trian::Triangulation,target_domain::DomainStyle)
-  change_domain(a.plus,DomainStyle(a),target_trian,target_domain)
+# If SkeletonCellFieldPair is evaluated we just pick the plus side parent
+function evaluate!(cache,f::SkeletonCellFieldPair,x::CellPoint)
+  _f, _x = _to_common_domain(f.plus.parent,x)
+  cell_field = get_data(_f)
+  cell_point = get_data(_x)
+  lazy_map(evaluate,cell_field,cell_point)
+end
+
+# fix for CellFieldAt{T}(parent::OperationCellField) giving right output
```

Comment:
Can we try to improve this comment? What is the key idea we want to say here?

I guess we want to say that:
 1. `CellFieldAt{T}(OperationCellField)` results in calling `CellFieldAt{T}(operands)`.
 2. Without this override, `CellFieldAt{T}(::SkeletonCellField)` results in XXX, which is not correct.
 3. Thus, we need this override.

Correct?

Reply:
Yes exactly Alberto.